hail/python/hail/linalg/blockmatrix.py

```diff
@@ -664,17 +670,30 @@ def write_from_entry_expr(entry_expr, path, overwrite=False, mean_impute=False,
                 field = Env.get_uid()
                 mt.select_entries(**{field: entry_expr})._write_block_matrix(path, overwrite, field, block_size)
         else:
-            n_cols = mt.count_cols()
             mt = mt.select_entries(__x=entry_expr)
-            mt = mt.select_rows(__count=agg.count_where(hl.is_defined(mt['__x'])),
-                                __sum=agg.sum(mt['__x']),
-                                __sum_sq=agg.sum(mt['__x'] * mt['__x']))
-            mt = mt.select_rows(__mean=mt['__sum'] / mt['__count'],
-                                __centered_length=hl.sqrt(mt['__sum_sq'] -
-                                                          (mt['__sum'] ** 2) / mt['__count']),
-                                __length=hl.sqrt(mt['__sum_sq'] +
-                                                 (n_cols - mt['__count']) *
-                                                 ((mt['__sum'] / mt['__count']) ** 2)))
+            compute = {
+                '__count': agg.count_where(hl.is_defined(mt['__x'])),
+                '__sum': agg.sum(mt['__x']),
+                '__sum_sq': agg.sum(mt['__x'] * mt['__x'])
+            }
+            if axis == 'rows':
+                n_cols = mt.count_cols()
+                mt = mt.select_rows(**compute)
+            else:
+                n_cols = mt.count_rows()
```

Comment:
Looks good, but I think the name `n_cols` is misleading here. Maybe something like `total_count`?

Reply:
Yeah, I wondered about that. `n_elements` maybe?

Reply:
I think that sounds good